src/Pages/login/login.jsx

```diff
@@ -0,0 +1,98 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+export default () => {
+  const navigate = useNavigate();
+  const [userName, setUserName] = useState(null);
+  const [password, setPassword] = useState(null);
+
+  const [isUsernameActivated, setUserNameActivated] = useState(false);
+  const [isPasswordActivated, setPasswordActivated] = useState(false);
+
+  const [userNameAlert, setUserNameAlert] = useState(null);
+  const [passwordAlert, setPasswordAlert] = useState(null);
+
+  function checkUsername () {
+    if (userName == null) {
+      setUserNameAlert("Please enter user name.");
+    } else {
+      setUserNameAlert(null);
+    }
+  };
```

Comment:
We are still missing the use of react hooks. For example this function can be wrapped with a `useCallback` hook to avoid all sorts of issues we could run into like performance degradations, unnecessary re-renders and computations etc.
```suggestion
  const checkUsername = useCallback(() => {
    if (userName == null) {
      setUserNameAlert("Please enter user name.");
    } else {
      setUserNameAlert(null);
    }
  }, [userName]);
```

Reply:
Fixed 0b3553b46184e591d0862bcbd05ffd4fdf5f3363